docs/6.x/pack.md

```diff
@@ -131,6 +131,59 @@ docker run -e OPS_URL=<opscenter url> \
     by setting `--state-dir`. You can use unique temporary directory
     to avoid sharing state between builds, or use parallel builds instead.
 
+## Vendoring
+
+When you execute the `tele build` command, `tele` discovers all Docker images
+referenced by your cluster or application image resources (Helm charts or plain
+Kubernetes spec files) and packages them inside the resulting cluster or
+application image tarball. This process is called "vendoring" and it gives the
+Gravity images their "self-sufficiency" property: all vendored images are
+synchronized with the in-cluster private Docker registry when the cluster is
+installed and Kubernetes pulls them from this local registry when spinning up
+pods.
+
+#### Private Docker Registries
+
+To be able to vendor a Docker image, it needs to be available via the local
+Docker client. To see a list of images for the local Docker client, you can
+typically use the `docker images` command.
+
+If an image being vendored is not available via the local Docker client, `tele`
+will attempt to pull it from a remote registry specified by the image, or the
+default [Docker Hub](https://hub.docker.com) if the registry is not specified.
+
+In case the image belongs to a private Docker registry, your local Docker client
+must be configured with proper credentials for it (e.g. via `docker login` or
+[TLS certificates](https://docs.docker.com/engine/security/certificates/)) in
+order for `tele` to be able to pull it.
+
+#### Image References Discovery
+
+The `tele` tool can extract image references from all core Kubernetes objects
+such as pods, deployments, replica and daemon sets and so on.
+
+!!! note "Helm chart images":
+    When building cluster or application image out of a Helm chart, `tele` will
+    render the Helm templates before extracting the image references. You can
+    use `--set` and `--values` flags to provide custom Helm values to `tele build`.
+
+For some objects, `tele` can't extract images because their schema is not known
```

Comment:
`tele` can only extract images from resource types it knows about. The `ImageSet` resource allows to list additional
image references which `tele` would otherwise not be able to extract (i.e. from custom resource types):